src/views/AddAddressLedger.vue

```diff
@@ -0,0 +1,112 @@
+<template>
+    <div class="container">
+        <SmallPage>
+            <LedgerUi v-if="state === 'ledger-interaction'"></LedgerUi>
+            <IdenticonSelector v-else-if="state === 'identicon-selection'" :accounts="addressesToSelectFrom"
+                               :confirmAccountSelection="false" @identicon-selected="_onAddressSelected">
+            </IdenticonSelector>
+            <Loader v-if="state === 'finished'" state="success" title="Address Added"></Loader>
+        </SmallPage>
+
+        <button class="global-close nq-button-s" @click="close">
+            <span class="nq-icon arrow-left"></span>
+            Back to {{request.appName}}
+        </button>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Vue } from 'vue-property-decorator';
+import { PageBody, SmallPage } from '@nimiq/vue-components';
+import LedgerUi from '../components/LedgerUi.vue';
+import Loader from '../components/Loader.vue';
+import IdenticonSelector from '../components/IdenticonSelector.vue';
+import { Static } from '../lib/StaticStore';
+import { ParsedSimpleRequest } from '../lib/RequestTypes';
+import { Address } from '../lib/PublicRequestTypes';
+import { WalletInfo } from '../lib/WalletInfo';
+import { AccountInfo } from '../lib/AccountInfo';
+import LedgerApi from '../lib/LedgerApi';
+import { WalletStore } from '../lib/WalletStore';
+import { ACCOUNT_MAX_ALLOWED_ADDRESS_GAP, ERROR_CANCELED } from '../lib/Constants';
+import LabelingMachine from '../lib/LabelingMachine';
+
+@Component({components: {PageBody, SmallPage, LedgerUi, Loader, IdenticonSelector}})
+export default class AddAddressLedger extends Vue {
+    private static readonly State = {
+        LEDGER_INTERACTION: 'ledger-interaction', // can be instructions to connect or also display of an error
+        IDENTICON_SELECTION: 'identicon-selection',
+        FINISHED: 'finished',
+    };
+
+    @Static private request!: ParsedSimpleRequest;
+
+    private state: string = AddAddressLedger.State.LEDGER_INTERACTION;
+    private account!: WalletInfo;
+    private addressesToSelectFrom: AccountInfo[] = [];
+
+    private async created() {
+        // called every time the router shows this page
+        const account = await WalletStore.Instance.get(this.request.walletId);
+        if (!account) {
+            this.$rpc.reject(new Error('Account ID not found'));
+            return;
+        }
+        this.account = account;
+
+        let startIndex = 0;
+        const newestAddress = Array.from(account.accounts.values()).pop();
+        if (newestAddress) {
+            const newestIndex = LedgerApi.getKeyIdForBip32Path(newestAddress.path);
+            startIndex = (newestIndex !== null ? newestIndex : -1) + 1;
+        }
+
+        const pathsToDerive = [];
+        for (let keyId = startIndex; keyId < startIndex + ACCOUNT_MAX_ALLOWED_ADDRESS_GAP; ++keyId) {
+            pathsToDerive.push(LedgerApi.getBip32PathForKeyId(keyId));
```

Comment:
I know this is not a problem with this PR specifically, but with the LedgerUi class: I think the last number in the derivation path should not be called `keyId`, as we already use that for the _ID_ of the key/wallet/account in the store. What you mean here is an _account index_ or _addresss index_. Emphasis on _index_, not _id_.

Reply:
Yes, several things in LedgerApi should be renamed now.
I noted that on my list, but didn't want to do it in this PR.